### PR TITLE
Python-specific conversions for SWIG

### PR DESCRIPTION
**Pull request by dmikhalin**
## What is the goal of this PR?

We define python-specific conversions for SWIG for arrays, exceptions and `ValueType`.

### CODE REVIEW

File: java/test/behaviour/typeql/language/insert/BUILD [hunk 1]
```diff
@@ -55,7 +55,7 @@ typedb_behaviour_java_test(
     runtime_deps = [
         "//java/test/behaviour/config:parameters",
     ],
-    size = "large",
+    size = "enormous",
```
Comment (flyingsilverfin) [inline java/test/behaviour/typeql/language/insert/BUILD:58]: Why has this gotten larger? Are the tests slower?

File: rust/swig/typedb_client_python.swg [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+}
+
+%pythoncode %{
+    TypeDBClientException = native_client_python.TypeDBClientException
+%}
+
+/* ValueType* needs special handling */
+%typemap(in) enum ValueType * {
+    if ($input != Py_None) {
```
Comment (flyingsilverfin) [inline rust/swig/typedb_client_python.swg:105]: How does swig translate `none`? Is this the protection we built for the issue we had found where we passed a null and it got mapped to `object` implicitly?

File: java/test/behaviour/typeql/language/insert/BUILD (hunk 1 above)
Comment (dmikhalin) [reply java/test/behaviour/typeql/language/insert/BUILD:58]: There was a TL at Factory (900 s).

File: rust/swig/typedb_client_python.swg (hunk 2 above)
Comment (dmikhalin) [reply rust/swig/typedb_client_python.swg:105]: The main problem was with all values except `None`. Now we allocate memory for all such `ValueType`s (and free it later), and the value remains `null` if it was `None` in python.

File: java/test/behaviour/typeql/language/insert/BUILD (hunk 1 above)
Comment (flyingsilverfin) [reply java/test/behaviour/typeql/language/insert/BUILD:58]: How close was it to time out before we changed to the rust-wrapped client? We should always know the root cause of why we're increasing test timeouts

File: rust/swig/typedb_client_python.swg (hunk 2 above)
Comment (flyingsilverfin) [reply rust/swig/typedb_client_python.swg:105]: ok, got it. i hope we don't introduce any memory leaks!!

File: java/test/behaviour/typeql/language/insert/BUILD (hunk 1 above)
Comment (dmikhalin) [reply java/test/behaviour/typeql/language/insert/BUILD:58]: When the previous PR was tested on Factory, this test lasted for 830 s. The last attempt for this PR took 757 s. So it is close to 900 and sometimes occasionally take longer.

Comment (flyingsilverfin) [reply java/test/behaviour/typeql/language/insert/BUILD:58]: Hmm ok, we can make it larger

**Pull request merged**